pynest/examples/csa_example.py

```diff
@@ -19,57 +19,96 @@
 # You should have received a copy of the GNU General Public License
 # along with NEST.  If not, see <http://www.gnu.org/licenses/>.
 
+'''
+Using CSA for connection setup
+------------------------------
+
+This example sets up a simple network in NEST using the Connection Set
+Algebra (CSA). The network uses random connectivity with a connection 
+probability of 10%.
+
+See also Djurfeldt M, Davison AP and Eppler JM (2014) **Efficient
+generation of connectivity in neuronal networks from
+simulator-independent descriptions**, *Front. Neuroinform.*
+http://dx.doi.org/10.3389/fninf.2014.00043
+'''
+
+'''
+First, we import necessary modules for simulation and plotting.
+'''
+
+import sys
 import nest
+from nest import voltage_trace
+from nest import visualization
+
+'''
+We now check for the availability of the CSA Python module and
+exit with an error message, if we don't have it.
+'''
 
 try:
     import csa
     haveCSA = True
 except ImportError:
-    haveCSA = False
+    print("This example requires CSA to be installed in order to run!")
+    sys.exit()
 
-try:
-    from PIL import Image
-    havePIL = True
-except ImportError:
-    havePIL = False
+'''
+We create a ``random`` connection set with a probability of 0.1
+and two values (10000.0 and 1.0) used as weight and delay,
+respectively.
+'''
 
-def csa_example():
+cs = csa.cset(csa.random(0.1), 10000.0, 1.0)
 
-    cs = csa.cset(csa.random(0.1), 10000.0, 1.0)
+'''
+The neurons of the pre- and postsynaptic populations are
```

Comment:
You are jumping back and forth between active "we create" and passive "neurons ... are created" throughout the text.